src/main/java/teammates/common/datatransfer/attributes/EntityAttributes.java

```diff
@@ -73,4 +74,66 @@ public void addNonEmptyError(String error, List<String> errors) {
 
         errors.add(error);
     }
+
+    /**
+     * Helper class to determine whether a field should be updated or not.
+     *
+     * <p>The class behaves like {@link java.util.Optional} but allows null value.
+     */
+    protected static class UpdateOption<T> {
+
+        private static final UpdateOption<?> EMPTY = new UpdateOption<>();
+
+        private boolean isValuePresent;
```

Comment:
Do you really need this? It looks to me that `this.isValuePresent = this.value == null`.